### ISSUE TITLE
_.pickBy adding extra undefined element in IE9

### ISSUE THREAD

**Issue by brendangibson** [lodash/lodash#2484]
Version: 4.13.1

If I run this code on IE9, I get an error saying that o is undefined.

```
		const bob = _.pickBy({me: {select: true}, you: {select: false}}, function (o) {
			console.log('o: ',o);
			return o.select;
		});

		console.log(bob);
```

On modern browsers there is no issue.

**Comment by jdalton** [comment]
`const` isn't supported in IE9. You might try simplying your code to something supported out of the box.

**Comment by brendangibson** [comment]
This does the same:

```
var bob = _.pickBy({me: {select: true}, you: {select: false}}, function (o) {
			console.log('o: ',o);
			return o.select;
		});

		console.log(bob);
```

(original code was transpiled with babel)

**Comment by jdalton** [comment]
Hmm I can't repro the error.

**Comment by brendangibson** [comment]
I found out the cause of my problem, and it's (mostly) not a problem with lodash.

The issue I was having was that the `object` parameter of `pickBy` had too many members when the predicate function was applied. 
The extra members turned out to be global `Symbols`. As IE9 does not support `Symbols`, and babel doesn't transpile them, and I still want to use them because I think they are nice, I used a browser polyfill.

That polyfill was creating `getOwnPropertySymbols`, so the IE check below, in lodash code, was not firing and *all* the symbols were being added to the object that the predicate function acted on. The symbols were added as `undefined`, leading to the error for which I created this issue.
 

```/**
     * Creates an array of the own enumerable symbol properties of `object`.
     *
     * @private
     * @param {Object} object The object to query.
     * @returns {Array} Returns the array of symbols.
     */
    function getSymbols(object) {

      // Coerce `object` to an object to avoid non-object errors in V8.
      // See https://bugs.chromium.org/p/v8/issues/detail?id=3443 for more details.
      return getOwnPropertySymbols(Object(object));
    }
    // Fallback for IE < 11.
    if (!getOwnPropertySymbols) {
      getSymbols = stubArray;
    }
```

On my end the problem is now fixed because I use a different polyfill. I am not sure why all the Symbols are added in lodash. I assume it is expected behaviour.
